haskell/test/Test/Ligo/BaseDAO/FA2.hs

```diff
@@ -31,88 +31,90 @@ import Lorentz.Test hiding (withSender)
 
 {-# ANN module ("HLint: ignore Reduce duplication" :: Text) #-}
 
 zeroTransferScenario :: MonadNettest caps base m => OriginateFn m -> m ()
 zeroTransferScenario originateFn = do
   nonexistent :: Address <- newAddress "nonexistent"
-  ((owner1, _), _, dao, _) <- originateFn
+  ((owner1, _), _, dao, _,  _) <- originateFn
   let params = [ FA2.TransferItem
         { tiFrom = nonexistent
         , tiTxs = [ FA2.TransferDestination
             { tdTo = owner1
-            , tdTokenId = unfrozenTokenId
+            , tdTokenId = frozenTokenId
             , tdAmount = 0
             } ]
         } ]
 
   withSender (AddressResolved nonexistent) $
     call dao (Call @"Transfer") params
 
+-- TODO: Add a test for transfer with an sender who is not admin.
 validTransferScenario :: MonadNettest caps base m => OriginateFn m -> m ()
 validTransferScenario originateFn = do
-  ((owner1, op1), (owner2, _), dao, _) <- originateFn
+  ((owner1, _op1), (owner2, _), dao, _, admin) <- originateFn
   let params = [ FA2.TransferItem
         { tiFrom = owner1
         , tiTxs = [ FA2.TransferDestination
             { tdTo = owner2
-            , tdTokenId = unfrozenTokenId
+            , tdTokenId = frozenTokenId
             , tdAmount = 10
             } ]
         } ]
-  withSender (AddressResolved op1) $ call
+  withSender (AddressResolved admin) $ call
     dao (Call @"Transfer") params
 
   -- Check the balance
   consumer <- originateSimple "consumer" [] contractConsumer
 
-  withSender (AddressResolved owner2) $ call dao (Call @"Balance_of")
-    (mkFA2View [ FA2.BalanceRequestItem
-      { briOwner = owner2
-      , briTokenId = unfrozenTokenId
-      } ] consumer)
-
-  checkStorage (AddressResolved $ toAddress consumer)
-    (toVal [[((owner2, 0 :: Natural), 110 :: Natural)]] )
-
-validTransferOwnerScenario :: MonadNettest caps base m => OriginateFn m -> m ()
-validTransferOwnerScenario originateFn = do
-  ((owner1, _), (owner2, _), dao, _) <- originateFn
-  let params = [ FA2.TransferItem
-        { tiFrom = owner1
-        , tiTxs = [ FA2.TransferDestination
-            { tdTo = owner2
-            , tdTokenId = unfrozenTokenId
-            , tdAmount = 10
-            } ]
-        } ]
-
-  withSender (AddressResolved owner1) $ call dao (Call @"Transfer") params
-  -- Check the balance
-  consumer <- originateSimple "consumer" [] contractConsumer
-
-  withSender (AddressResolved owner2) $ call dao (Call @"Balance_of")
+  withSender (AddressResolved admin) $ call dao (Call @"Balance_of")
     (mkFA2View [ FA2.BalanceRequestItem
       { briOwner = owner2
-      , briTokenId = unfrozenTokenId
+      , briTokenId = frozenTokenId
       } ] consumer)
 
   checkStorage (AddressResolved $ toAddress consumer)
-    (toVal [[((owner2, 0 :: Natural), 110 :: Natural)]] )
+    (toVal [[((owner2, frozenTokenId), 110 :: Natural)]] )
+
+-- Disabling this test since there is no unfrozen token available to test it.
```

Comment:
Please add a `TODO #54` on kind of tests, as they'll need to be reactivated by that issue.

Reply:
Done.